src/sources/soundsourcesndfile.cpp

```diff
@@ -31,21 +31,25 @@ SoundSource::OpenResult SoundSourceSndFile::tryOpen(const AudioSourceConfig& /*a
     m_pSndFile = sf_open(getLocalFileName().toLocal8Bit(), SFM_READ, &sfInfo);
 #endif
 
-    if (!m_pSndFile) {   // sf_format_check is only for writes
-        qWarning() << "Error opening libsndfile file:" << getUrlString()
-                << sf_strerror(m_pSndFile);
-        return OpenResult::FAILED;
-    }
-
     switch (sf_error(m_pSndFile)) {
     case SF_ERR_NO_ERROR:
+        DEBUG_ASSERT(m_pSndFile != nullptr);
         break; // continue
     case SF_ERR_UNRECOGNISED_FORMAT:
         return OpenResult::UNSUPPORTED_FORMAT;
     default:
-        qWarning() << "Error opening libsndfile file:" << getUrlString()
-                << sf_strerror(m_pSndFile);
-        return OpenResult::FAILED;
+        const QString errorMsg(sf_strerror(m_pSndFile));
+        if (errorMsg.toLower().indexOf("unknown format") != -1) {
+            // NOTE(uklotzde 2016-05-11): This actually happens when
+            // trying to open a file with a supported file extension
+            // that contains data in an unsupported format!
+            // Example: m4a + MPEG4/AAC
+            return OpenResult::UNSUPPORTED_FORMAT;
+        } else {
+            qWarning() << "Error opening libsndfile file:" << getUrlString()
+                    << sf_strerror(m_pSndFile);
```

Comment:
Ups, of course, that was actually my intention! Fixed.

On 05/11/2016 08:30 PM, RJ Ryan wrote:
>
> In src/sources/soundsourcesndfile.cpp 
> <https://github.com/mixxxdj/mixxx/pull/948#discussion_r62901374>:
>
> >          break; // continue
> >      case SF_ERR_UNRECOGNISED_FORMAT:
> >          return OpenResult::UNSUPPORTED_FORMAT;
> >      default:
> > -        qWarning() << "Error opening libsndfile file:" << getUrlString()
> > -                << sf_strerror(m_pSndFile);
> > -        return OpenResult::FAILED;
> > +        const QString errorMsg(sf_strerror(m_pSndFile));
> > +        if (errorMsg.toLower().indexOf("unknown format") != -1) {
> > +            // NOTE(uklotzde 2016-05-11): This actually happens when
> > +            // trying to open a file with a supported file extension
> > +            // that contains data in an unsupported format!
> > +            // Example: m4a + MPEG4/AAC
> > +            return OpenResult::UNSUPPORTED_FORMAT;
> > +        } else {
> > +            qWarning() << "Error opening libsndfile file:" << getUrlString()
> > +                    << sf_strerror(m_pSndFile);
>
> can you re-use errorMsg here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/948/files/9d5718c22d63dd01de5895449cfe76958d51fb49#r62901374>
>